### ISSUE TITLE
Officially support macOS

### ISSUE THREAD

**Issue by JackRichardBusiness** [firebase/firebase-ios-sdk#583]
I would like macOS to be officially supported by Firebase.

**Comment by morganchen12** [comment]
I think this won't be a priority for a while. @salqadri can maybe talk more about it.

**Comment by salqadri** [comment]
Hey Jack,

We actively encourage and assist the community in building and maintaining the macOS SDK. Could you tell us more on why it's important to you for macOS to be 'officially supported'?

**Comment by JackRichardBusiness** [comment]
A macOS app I am making requires Firebase Database, and an officially
supported version would be more stable.

**Comment by paulb777** [comment]
@JackRichardBusiness Will you be more specific about where the community supported macOS SDK hasn't met your stability requirements?  Any [issues](https://github.com/firebase/firebase-ios-sdk/issues) that you're seeing?

**Comment by morganchen12** [comment]
Closing since this isn't going to happen soon and this issue hasn't garnered much community support.  We can revisit this in the future.

**Comment by joelgetaction** [comment]
For the reference of others, did you mean this?  https://github.com/firebase/firebase-ios-sdk#macos

**Comment by paulb777** [comment]
@joelgetaction This - https://github.com/firebase/firebase-ios-sdk#macos-and-tvos

**Comment by jhoughjr** [comment]
Bump. There should be no reason this isn't usable for both platforms.

**Comment by rismay** [comment]
Can we add support for macOS via Carthage? Right now we only have this: 
`binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseAnalyticsBinary.json"`

**Comment by paulb777** [comment]
@rismay We'll investigate. In the meantime, if you want a binary framework version of a mac distribution, [CocoaPods/Rome](https://github.com/firebase/firebase-ios-sdk/blob/master/Rome.md) is an option.

**Comment by eebean2** [comment]
Can we get support for this since Apple is launching cross platform support for iOS apps?

**Comment by appfrilans** [comment]
I agree with @eebean2. A lot of iPad apps will be ported coming month due to Apples new Catalyst project. One switch in Xcode will make your app into a macOS app. However, frameworks need to support macOS and the only one I got an issue with was Firebase SDK!

**Comment by paulb777** [comment]
We're digesting all of the WWDC updates and will reassess the Firebase multi-platform official support story.

In the meantime, please file any specific issues you have with the current community macOS support.

**Comment by guidedways** [comment]
Not to mention there's very little multi-platform solutions out there. Can't think of anything other than realm database. Having Firebase on macOS would be a game changer for all of us - we'd love to switch over to Firebase wholeheartedly if given the chance.

**Comment by cliftonlabrum** [comment]
If you can include Firebase in an iPadOS app, won’t it work if deployed on a Mac? It’ll be running on an iOS-built app, won’t it?

**Comment by EliteTechnicalCare** [comment]
> If you can include Firebase in an iPadOS app, won’t it work if deployed on a Mac? It’ll be running on an iOS-built app, won’t it?

Unfortunately it is not working at the moment. It needs to be able to run on macOS

**Comment by guidedways** [comment]
Even if it did work for Catalyst / iPadOS apps on macOS, we'd appreciate proper support for native macOS app. Just please do it and get it done and over with so we can all enjoy cross platform deployment of Firebase in our apps.

**Comment by appfrilans** [comment]
Is it possible to include firebase for macOS (UIKit for Mac) via pod 'FirebaseCore' ? I tried to make a separate target and set it in its own CocoaPods target (platform :macos, '10.15'), but importing via "import FirebaseCore" just gives me an error:

No such module 'FirebaseCore'

Any ideas? Or is this way impossible?

**Comment by Umar-M-Haroon** [comment]
So far I’ve had some luck getting support for macOS through Catalyst. Here’s what I did

In the same project that works on iOS, select the Mac checkbox

Ran pod install

Solved certificates issues with the gRPC by selecting the team as needed (Only needed if Firestore pod is installed)

Selected the Mac as the target and commented any code pertaining to UIWebView in the FIRAuth files (only needed if FIRAuth pod is installed)

Also removed “AssetsLibrary” -framework from the .xcconfigs using the find and replace feature

**Comment by EliteTechnicalCare** [comment]
> Selected the Mac as the target and commented any code pertaining to UIWebView in the FIRAuth files (only needed if FIRAuth pod is installed)

Any adverse effects on FIRAuth?

**Comment by Umar-M-Haroon** [comment]
I haven't had any issues with FIRAuth. If your deployment target is greater than 7.0 you should be fine.

**Comment by OlexandrStepanov** [comment]
Hi folks,

I'm wondering, is FirebaseAnalytics support for macOS somewhere on Google's radar? Are there any ETA for this feature available for macOS apps?
This would be a huge improvement, especially because there used to be Fabric analytics, which is now migrating to Firebase, and analytics simply kicked off.

With regards.

**Comment by paulz** [comment]
Mac Catalyst build fails for me:
```
ld: in .../Pods/FirebaseAnalytics/Frameworks/FIRAnalyticsConnector.framework/FIRAnalyticsConnector(FIRConnectorUtils_d79571aba36a7d46e5c6ca87a6fec1c1.o), 
building for Mac Catalyst, but linking in object file built for iOS Simulator, file 
.../Pods/FirebaseAnalytics/Frameworks/FIRAnalyticsConnector.framework/FIRAnalyticsConnector' 
for architecture x86_64
```

when including Firebase Analytics:
```
pod 'Firebase/Analytics'
```

**Comment by paulb777** [comment]
As documented at https://github.com/firebase/firebase-ios-sdk#tvos-macos-and-catalyst, we do not currently support Analytics for Catalyst (#4563)

**Comment by paulz** [comment]
Looking for a workaround, how can I still use Analytics on iOS and exclude it for the Catalyst build

**Comment by paulb777** [comment]
@paulz See https://github.com/firebase/firebase-ios-sdk/issues/4563#issuecomment-573281457

**Comment by andytriboletti** [comment]
That workaround didn't work for me.

**Comment by hannaanderssonpdx** [comment]
Deadline for Fabric’s demise is round the corner and we still don’t have proper macOS support. We heavily rely on analytics for our Mac apps and that’s all going away soon. If fabric can do it, how difficult is it to get it ready for Firebase? Not to mention I’m already annoyed Firebase requires a lot more clicks for what was easy to get to in Fabric.

**Comment by cliftonlabrum** [comment]
I think official and ongoing support for macOS is a lost cause. I’ve liked Microsoft App Center as a replacement for crash logging and analytics. It’s worth checking out: https://appcenter.ms

**Comment by hannaanderssonpdx** [comment]
@cliftonlabrum thanks for that, didn’t realize there was an alternative! What a shame. Google keeps emailing us threats that Fabric is discontinuing and at the same time hasn’t given us something to port to. I prefer keeping everything under the same portal so will move all the iOS + Android + macOS apps to Microsoft now.

**Comment by hannaanderssonpdx** [comment]
@paulb777 can we at least have an official statement on why macOS support for analytics / performance and Firebase database is being delayed? Why do you guys keep asking why official support is needed. Isn’t it obvious? Analytics aren’t available. That’s 50% of what crashlytics used to offer.

**Comment by guidedways** [comment]
Surely someone clicks on the “Send” button for all these “Fabrics is shutting down” emails we keep receiving. Ironically it asks us to migrate to Firebase, which does not support macOS apps. Why acquire a service and then “lame” it down? Google should have left fabric alone.

**Comment by andytriboletti** [comment]
I had to create a git branch for my Catalyst and iOS project with the Analytics pod commented out on the catalyst branch. I tried creating multiple targets with the analytics pod only in the iOS target, but couldn't deploy to the app store because of a duplicated library problem. Thanks for the Microsoft tip. I will check it out.

**Comment by guidedways** [comment]
I’ve received a response from Google Developer support that Firebase will not support analytics for macOS anytime soon. Okay I’m out of here.

**Comment by andytriboletti** [comment]
> Looking for a workaround, how can I still use Analytics on iOS and exclude it for the Catalyst build

I got this working:
https://stackoverflow.com/a/60213321/211457

**Comment by SebastianBO** [comment]
@andytriboletti nice!! Sick timing to just come back - will test it out

**Comment by shobhitchugh** [comment]
Hi everyone. This is Shobhit, Product Manager for Crashlytics. Thank you for the feedback. 
 
Firebase does not officially support Catalyst, macOS, and tvOS apps. If you are using Fabric Crashlytics with macOS and tvOS apps, you can migrate these apps to Firebase as iOS apps. If you have an iOS app with the same bundle ID that already exists in Firebase, then you will have to create your macOS/tvOS app in a separate Firebase Project. 
 
The Crashlytics SDK will continue to report crashes for Catalyst, macOS, and tvOS apps. However, features that depend on Google Analytics or other Firebase products (i.e., Crash-Free Users, Velocity Alerts, and breadcrumbs) will not work for these apps.
 
We have no timeline for adding Firebase support for macOS and tvOS at this time but are continually evaluating platforms and will keep it on our radar.

**Comment by guidedways** [comment]
@shobhitchugh I hope you appreciate this is extremely frustrating for all of us. Analytics works just fine within Fabric, yet we're being forced out of it to embrace a broken platform, that does not support Analytics, Events and and fails to report simple things such as "Launch Status for build xxx" → something that was incredibly useful in Crashlytics (Fabric). One click and you can see the growing number of users / crashes. Not only is there no macOS support for this in Firebase, even the iOS version of Analytics does not provide us with this simple stat (I know "Latest release" does a lot more - exactly, it does a lot MORE and it's annoyingly slow and frustrating to use).

If google, with all its resources, cannot completely support an already working system, it should have left it alone. I realize the need to deprecate Fabric and the switch to Firebase, but why not keep it around for longer, especially macOS apps? We're going to be crippled once it's turned off after 31st March.

**Comment by jcampbell05** [comment]
We are adding macOS support as we need it for our new launch but we will have to abandon firebase which is a shame as we've been customers of it for many many year

**Comment by paulb777** [comment]
@jcampbell05 Sorry to hear that. 

Is the issue that the support is not "official" or that the "community" support is missing features that you need?

**Comment by guidedways** [comment]
@paulb777 I don't think anyone would really care if it were "official" as long as we all got what we've been asking for, for years? And that primarily is: "Analytics" support for macOS. We should be able to see "Analytic Events" as we do for our iOS apps. This feature works just fine in Fabric and I don't understand how or why it's difficult to port this over to Firebase. If there are technical hurdles, why not keep Fabric running and supporting macOS apps only till "official" support arrives in the year 3020 - oh wait, by then it wouldn't matter since those of us on this thread won't be alive.

We want Analytics to work for macOS. We want to be able to see the number of users using the app, other than just crash reports. We want to be able to see all the custom events we already are able to track in Fabric, for macOS.

**Comment by paulb777** [comment]
@guidedways I'm familiar with the reasons for `Analytics` macOS support. However, @jcampbell05 said that they'd been Firebase customers for many years - which wouldn't apply to the Fabric migration reason for Analytics - since Crashlytics is a relatively recent addition to Firebase.

Let's have the Analytics specific discussion in #1732 and leave this issue for discussion around converting the existing Firebase community support for macOS to official support (which I understand is dependent upon the more specific #1732).

**Comment by jcampbell05** [comment]
> Is the issue that the support is not "official" or that the "community" support is missing features that you need?

I'm a long time user of Fabric and Firebase to clarify.

Community support would be better than nothing if it mean't we get the analytics features from Fabric/Google Analytics as well as GCM support.

But ultimately we would need some sort of official support as we don't want to be relying on an SDK that is only being supported by people in their spare time. 

We're looking into adding a desktop companion to our iOS app so adding officially supported port for catalyst would be great.

Luckily all of the code here is open-source so if push comes to shove we can do it ourselves but then the issue is since there isn't "official" support - will the Firebase dashboard work well with this or does it assume all apps are either mobile or web and not catalyst.

For this application we would have to look in the longer term for cross-platform solutions.

I'm aware one blocker is you are waiting on some cocoa pods support but that may be a long time coming - instead I would suggest you do as Stripe are doing and bundle a Xcframework. 

We would be happy to use that for the time being instead of via Cocoapods if it mean't we can use the framework everywhere we need.

**Comment by guidedways** [comment]
> Let's have the Analytics specific discussion in #1732 and leave this issue for discussion around converting the existing Firebase community support for macOS to official support (which I understand is dependent upon the more specific #1732).

@paulb777  Fair enough. In that case, yes, we would (all?) like macOS to be **officially** supported by Firebase Crashlytics / Answers / Analytics etc - please 😀

**Comment by guidedways** [comment]
@jcampbell05 Probably worth leaving a +1 under #1732, too

**Comment by cbenhagen** [comment]
Please upvote the issue instead of leaving +1 comments which send a notification to everyone subscribed. Thanks!

**Comment by dtnam1997** [comment]
Please fill in the following fields:
Pre-built SDK from the website or open-source from this repo:
Firebase C++ SDK version: 6.12.0
Firebase plugins in use (Auth, Database, etc.): Auth, Realtime
Additional SDKs you are using (Facebook, AdMob, etc.): �Facebook
Platform you are using the C++ SDK on (Mac, Windows, or Linux): MAC
Platform you are targeting (iOS, Android, and/or desktop): MAC OS

Please describe the issue here:
Hi firebase team!

Currently, I'm querying the data on the MAC OS platform. I'm facing the problem as follows :

The query takes a long time (1 to 3 minutes) to return a response. So This affects the user experience of my app.

On the IOS, Android and Windows platforms all work very well. I only have this problem on MAC OS.

I also tested both link options x86_64 and universal. But this also could not help me solve this problem.

Here is the information regarding my project :

Xcode: 11.3.1

Firebase_cpp_sdk: [ as above ]

My query : ( same as on the IOS, Android and Windows platforms)

auto user = ap::Database::getInstance().getUserInfo();

firebase::database::DatabaseReference dbref = database->GetReference();

std::string path = cocos2d::StringUtils::format("users/%d/%s/", user._id, USER_REALTIME_VARIABLE);

auto result = dbref.Child(path).GetValue();

result.OnCompletion([callback](const Future<firebase::database::DataSnapshot>& result_data) {

	//my codes

});

As I know, MACOS doesn't officially support from firebase team, right?

Hopefully, I can get answers soon from your side.

**Comment by paulb777** [comment]
@dtnam1997 I'm deleting your post above since it is cluttering a popular issue in the wrong repo.  Please follow it at https://github.com/firebase/firebase-cpp-sdk/issues/50

**Comment by itsdamslife** [comment]
I am building a macOS standalone process, a command line tool binary (non UI app). I am trying to integrate crashlytics into it. I manage to fix all other issues with bundle identifier, fetching dsym files and uploading etc. 

The crash reports are getting uploaded to firebase console as per network proxy logs. I can see 202 Accepted status for POST requests to https://reports.crashlytics.com domain calls.

But I am not seeing any reports in firebase console. 

Note: I also have another app under same project which is an UI based macOS app (.app like iOS). For this app the crash reports are displayed perfectly in the dashboard. Only for command line binaries it does not show up in the console dashboard.

Please help.

**Comment by makuchaku** [comment]
Hi everyone,

I am Mayank, a Product Manager at Firebase.

I am excited to share that we are in the process of improving Firebase SDKs to support additional Apple platforms.

If you are already using Firebase in your app which supports multiple Apple platforms (tvOS/macOS/Catalyst/watchOS) or are looking to add support for those platforms, we'd love to get your feedback.

If you're interested in chatting with us and getting early access to future updates, please sign up using this form : https://forms.gle/odseGnshb3Jq8wuz5

Once approved, we will share further information on how to proceed.

Thank you!
Mayank (on behalf of Firebase SDK team)

**Comment by tnmendes** [comment]
Any update? The last message was 10 Feb and we are now 27 Sep

**Comment by morganchen12** [comment]
@tnmendes, early access has been going well with the partners who enrolled and we'll have more updates soon.
